lib/library.rb

```diff
@@ -0,0 +1,64 @@
+class Library
+  attr_reader :name,
+              :books,
+              :checked_out_books,
+              :authors,
+              :checkouts
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @checked_out_books = []
+    @authors = []
+    @checkouts = {}
+  end
+
+  def add_author(author)
+    @authors << author
+    author.books.each do |book|
+      books << book
+    end
+  end
+
+  def publication_time_frame_for(author)
+    max = 0
+    min = 10_000
```

Comment:
hmm, interesting to have an arbitrary value as the "min" here though. You'd have a serious case of Y2K when the year changes into 5 digits lol.